Doc/library/importlib.rst

```diff
@@ -420,30 +420,34 @@ ABC hierarchy::
         reloaded):
 
         - :attr:`__name__`
-            The name of the module.
+            The module's fully-qualified name.
 
         - :attr:`__file__`
-            The path to where the module data is stored (not set for built-in
-            modules).
+            The location of the module's source code.
+            It is ``None`` for namespace packages and not set for built-in
+            modules and frozen modules.
 
         - :attr:`__cached__`
-            The path to where a compiled version of the module is/should be
-            stored (not set when the attribute would be inappropriate).
+            The location of the module's compiled code.
+            It is not set for built-in modules, frozen modules
+            and namespace packages.
 
         - :attr:`__path__`
-            A list of strings specifying the search path within a
-            package. This attribute is not set on modules.
+            The iterable (possibly empty) of the locations of the module's
+            submodules.
+            It is not set for non-package modules.
 
         - :attr:`__package__`
-            The parent package for the module/package. If the module is
-            top-level then it has a value of the empty string. The
-            :func:`importlib.util.module_for_loader` decorator can handle the
-            details for :attr:`__package__`.
+            The module's ``__name__`` attribute for a package, the parent
```

Comment:
Also, technically, `pkg.__name__` isn't technically right, since it folks may modify that. :)  It is actually derived from `spec.__name__` for the module.

Reply:
Okay (I think you meant `__spec__.name`).